tests/zquantum/qaoa/estimator_test.py

```diff
@@ -42,76 +41,51 @@ def custom_run_circuit_and_measure(circuit, n_samples):
         backend.run_circuit_and_measure = custom_run_circuit_and_measure
         return backend
 
-    @pytest.fixture()
-    def n_samples(self, request):
-        return 10
-
-    def test_get_estimated_expectation_values_returns_expectation_values(
-        self, estimator, backend, circuit, target_operator, alpha
-    ):
-        value = estimator.get_estimated_expectation_values(
-            backend=backend,
-            circuit=circuit,
-            target_operator=target_operator,
-            alpha=alpha,
-        )
-        # Then
-        assert type(value) is ExpectationValues
-
     def test_raises_exception_if_operator_is_not_ising(
-        self, estimator, backend, circuit, alpha
+        self, estimator, backend, circuit
     ):
         # Given
-        target_operator = QubitOperator("X0")
+        estimation_tasks = [EstimationTask(QubitOperator("X0"), circuit, 10)]
         with pytest.raises(TypeError):
-            value = estimator.get_estimated_expectation_values(
+            estimator(
                 backend=backend,
-                circuit=circuit,
-                target_operator=target_operator,
-                alpha=alpha,
-            ).values
+                estimation_tasks=estimation_tasks,
+            )
 
     def test_cvar_estimator_raises_exception_if_alpha_less_than_0(
-        self, estimator, backend, circuit, target_operator
+        self, estimator, backend, estimation_tasks
     ):
-        alpha = -1
+        estimator.alpha = -1
         with pytest.raises(ValueError):
-            value = estimator.get_estimated_expectation_values(
+            estimator(
                 backend=backend,
-                circuit=circuit,
-                target_operator=target_operator,
-                alpha=alpha,
-            ).values
+                estimation_tasks=estimation_tasks,
+            )
 
     def test_cvar_estimator_raises_exception_if_alpha_greater_than_1(
-        self, estimator, backend, circuit, target_operator
+        self, estimator, backend, estimation_tasks
     ):
-        alpha = 2
+        estimator.alpha = 2
         with pytest.raises(ValueError):
-            value = estimator.get_estimated_expectation_values(
+            estimator(
                 backend=backend,
-                circuit=circuit,
-                target_operator=target_operator,
-                alpha=alpha,
-            ).values
+                estimation_tasks=estimation_tasks,
+            )
 
-    def test_cvar_estimator_returns_correct_values(
-        self, estimator, backend, target_operator, alpha
-    ):
+    def test_cvar_estimator_returns_correct_values(self, estimator, backend, operator):
```

Comment:
I think making these tests more robust is important. Either that or explicit tests of `_calculate_expectation_value_for_distribution` even though it's private since I can see lots of possible issues arising such as ordering of the bitstrings causing impact on the calculated values or improperly calculating the cutoff point.

Reply:
It'll be way easier to make them more robust when we migrate from `MockBackend` to `SimpleBackend`.

Reply:
Fair point